eth/watchers/eventdecoder_test.go

```diff
@@ -0,0 +1,74 @@
+package watchers
+
+import (
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/livepeer/go-livepeer/eth/contracts"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestEventDecoder_FindEventType(t *testing.T) {
+	dec, err := NewEventDecoder(stubBondingManagerAddr, contracts.BondingManagerABI)
+	require.Nil(t, err)
+
+	assert := assert.New(t)
+
+	// Test unknown contract address
+	log := newStubBaseLog()
+	log.Address = common.BytesToAddress([]byte("foo"))
+
+	_, err = dec.FindEventName(log)
+	assert.EqualError(err, "log not from known contract")
+
+	// Check that bondingManagerDecoder is used
```

Comment:
We should update this comment. Perhaps something like "Check for known contract address"

Reply:
fixed in ca5b5af